site/docs/uk/hosting/vps.md

```diff
@@ -52,130 +152,631 @@ systemd --- це потужний менеджер служб, який пере
+
+На сервері в терміналі виконайте наступну команду:
 
 ```sh
-systemctl status <назва-служби>
+pm2 startup
 ```
 
-#### Перезавантаження
+Вам буде надано команду, яку ви маєте виконати, щоб PM2 автоматично запрацював після перезавантаження серверу.
+
+Після цього виконайте ще одну команду:
 
 ```sh
-systemctl restart <назва-служби>
+pm2 save
+```
+
+Ця команда збереже список поточних застосунків, щоб запустити їх після перезавантаження сервера.
+
+Якщо ви створили новий застосунок і хочете його також зберегти, просто виконайте `pm2 save` ще раз.
+
+## Запуск бота на вебхуках
+
+Для запуску бота на вебхуках, вам потрібно буде скористатися серверним фреймворком і **не** викликати `bot.start()`.
+
+Ось приклад коду для запуску бота на вебхуках, який слід додати до основного файлу бота:
+
+::: code-group
+
+```ts [Node.js]
+import { webhookCallback } from "grammy";
+import { fastify } from "fastify";
+
+const server = fastify();
+
+server.post(`/${bot.token}`, webhookCallback(bot, "fastify"));
+
+server.listen();
 ```
 
-#### Зупинення
+```ts [Deno]
+import { webhookCallback } from "https://deno.land/x/grammy/mod.ts";
+
+const handleUpdate = webhookCallback(bot, "std/http");
+
+Deno.serve(async (req) => {
+  if (req.method === "POST") {
+    const url = new URL(req.url);
+    if (url.pathname.slice(1) === bot.token) {
+      try {
+        return await handleUpdate(req);
+      } catch (err) {
+        console.error(err);
+      }
+    }
+  }
+  return new Response();
+});
+```
+
+:::
+
+### Оренда домену
+
+Щоб звʼязати бота, який працює на вебхуках, із зовнішнім світом, потрібно придбати домен.
+
+Перейдіть на [сторінку пошуку доменних імен](https://hostinger.com.ua/kupyty-domen).
+У полі введення тексту введіть доменне імʼя виду `<назва>.<зона>`.
+Наприклад, `example.com`.
+
+Якщо бажаний домен вільний, натисніть кнопку "Додати" поряд із ним.
+Вас буде автоматично перенаправлено на сторінку оформлення замовлення, де ви також одразу зареєструєтеся у Hostinger, якщо досі не зареєстровані.
+Оплатіть домен.
+
+Щоб підключити бота, який працює на вебхуках, до зовнішнього світу, вам потрібно придбати домен.
+Ми знову будемо пояснювати це на прикладі Hostinger, але є багато інших сервісів, і всі вони працюють аналогічно.
+
+Перейдіть на [сторінку пошуку доменного імені](https://hostinger.com.ua/kupyty-domen).
+У полі для введення тексту введіть доменне ім'я у вигляді `<ім'я>.<зона>`.
+Наприклад, `example.com`.
+
+Якщо потрібний домен вільний, натисніть кнопку "Додати" поруч з ним.
+Ви будете автоматично перенаправлені на сторінку оформлення замовлення, де також відразу ж зареєструєтесь у Hostinger, якщо ви ще не зареєстровані.
+Оплатіть домен.
+
+#### Направлення домена на VPS
+
+Перш ніж ваш домен зможе працювати з вашим VPS, вам потрібно направити домен на ваш сервер.
+Для цього в [панелі керування Hostinger](https://hpanel.hostinger.com) натисніть кнопку "Керування" навпроти вашого домену.
```

Comment:
```suggestion
Для цього в [панелі керування Hostinger](https://hpanel.hostinger.com) натисніть кнопку "Керування".
```

Reply:
Hostinger has a really clear interface, but it is worth leaving this tip.